python/cross_service/resilient_service/runner.py

```diff
@@ -157,108 +151,94 @@ def deploy(self):
-                        print(
-                            f"\t\t{target['TargetHealth']['Reason']}: {target['TargetHealth']['Description']}\n"
+                        logging.warning(
+                            "%s: %s",
+                            target["TargetHealth"]["Reason"],
+                            target["TargetHealth"]["Description"],
                         )
-                print(
-                    f"\nNote that it can take a minute or two for the health check to update\n"
-                    f"after changes are made.\n"
+                logging.info(
+                    "Note that it can take a minute or two for the health check to update."
                 )
             elif choice == 2:
-                print("\nOkay, let's move on.")
-                print("-" * 88)
+                logging.info("Proceeding to the next part of the demo.")
 
-    def demo(self):
+    def demo(self) -> None:
+        """
+        Runs the demonstration, showing how the service responds to different failure scenarios
+        and how a resilient architecture can keep the service running.
+        """
         ssm_only_policy = f"{self.resource_path}/ssm_only_policy.json"
 
-        print("\nResetting parameters to starting values for demo.\n")
+        logging.info("Resetting parameters to starting values for the demo.")
         self.param_helper.reset()
 
-        print(
-            "\nThis part of the demonstration shows how to toggle different parts of the system\n"
-            "to create situations where the web service fails, and shows how using a resilient\n"
-            "architecture can keep the web service running in spite of these failures."
-        )
-        print("-" * 88)
-
-        print(
-            "At the start, the load balancer endpoint returns recommendations and reports that all targets are healthy."
+        logging.info(
+            "Starting demonstration of the service's resilience under various failure conditions."
         )
         self.demo_choices()
 
-        print(
-            f"The web service running on the EC2 instances gets recommendations by querying a DynamoDB table.\n"
-            f"The table name is contained in a Systems Manager parameter named '{self.param_helper.table}'.\n"
-            f"To simulate a failure of the recommendation service, let's set this parameter to name a non-existent table.\n"
+        logging.info(
+            "Simulating failure by changing the Systems Manager parameter to a non-existent table."
         )
         self.param_helper.put(self.param_helper.table, "this-is-not-a-table")
-        print(
-            "\nNow, sending a GET request to the load balancer endpoint returns a failure code. But, the service reports as\n"
-            "healthy to the load balancer because shallow health checks don't check for failure of the recommendation service."
-        )
+        logging.info("Sending GET requests will now return failure codes.")
```

Comment:
I don't think this is appropriate information because it's obvious. Is your concern that you want to use this opportunity to educate the user on terminology (shallow vs deep health checks)?

Reply:
Yes, exactly. When running the previous version and then this version, I find this version much less clear on what is going on and what I should expect to see.